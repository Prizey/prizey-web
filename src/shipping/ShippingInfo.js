import React from 'react'
import { New } from 'croods'
import { Button, Typography } from '@material-ui/core'
import Layout from 'design/Layout/Layout'
import { Redirect } from '@reach/router'

import UserForm from 'auth/UserForm'

export const ShippingForm = ({
  currentUser,
  location,
  create,
  creating,
  error,
}) => (
  <Layout location={location} currentUser={currentUser}>
    <Typography align="center" variant="h5">
      Shipping Address
    </Typography>
    <UserForm
      onSubmit={create}
      submitting={creating}
      submitError={error}
      user={currentUser}
      renderButton={() => (
        <Button variant="contained" color="primary" type="submit" fullWidth>
          SHIP IT!
        </Button>
      )}
    />
  </Layout>
)

export default ({ currentUser, location, setCurrentUser }) =>
  currentUser ? (
    <New
      name="users"
      path="/auth"
      method="PUT"
      render={props => (
        <ShippingForm
          {...props}
          currentUser={currentUser}
          location={location}
        />
      )}
      renderCreated={response => {
        setCurrentUser(response.data)
        return <Redirect to="/shipping-info/confirm" noThrow />
      }}
    />
  ) : (
    <Redirect to="/sign-in?next=/game" noThrow />
  )
