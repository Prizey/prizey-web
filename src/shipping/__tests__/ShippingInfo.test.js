import React from 'react'
import renderer from 'react-test-renderer'
import ShippingInfo from '../ShippingInfo'

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New -
      <div>
        render -
        {props.render({
          create: jest.fn(),
          creating: false,
          error: null,
        })}
      </div>
      <div>
        renderCreated - {props.renderCreated && props.renderCreated({ id: 1 })}
      </div>
    </div>
  ),
}))

jest.mock('redux-form', () => ({
  Field: props => <div {...props}>Field</div>,
  reduxForm: () => Component => props => (
    <Component handleSubmit={() => jest.fn()} {...props} />
  ),
}))

it('renders correctly', () => {
  const params = {
    currentUser: true,
  }
  const tree = renderer.create(<ShippingInfo {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('redirects correctly', () => {
  const params = {
    currentUser: false,
  }
  const tree = renderer.create(<ShippingInfo {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})
