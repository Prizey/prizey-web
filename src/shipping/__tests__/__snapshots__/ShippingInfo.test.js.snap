// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirects correctly 1`] = `null`;

exports[`renders correctly 1`] = `
<div
  method="PUT"
  name="users"
  path="/auth"
  render={[Function]}
  renderCreated={[Function]}
>
  New -
  <div>
    render -
    <div
      className="Component-root-8"
    >
      <div
        className="Component-header-6"
      >
        <a
          aria-label="My profile"
          className="MuiButtonBase-root-16 MuiIconButton-root-10 Component-root-9"
          href="/profile"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiIconButton-label-15"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-19 MuiSvgIcon-fontSizeLarge-27"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <g>
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33C4.62 15.49 4 13.82 4 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83z"
                />
                <path
                  d="M12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
                />
              </g>
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-91"
          />
        </a>
      </div>
      <div
        className="Component-container-3"
      >
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <img
            alt="Prizey"
            className="Component-logoImage-7"
            src="/logo.png"
          />
        </a>
        <div
          className="Component-caption-1"
        />
        <div
          className="Component-children-2"
        >
          <h5
            className="MuiTypography-root-28 MuiTypography-h5-44 MuiTypography-alignCenter-51"
          >
            Shipping Address
          </h5>
          <div
            initialValues={true}
            onSubmit={[MockFunction]}
            render={[Function]}
          >
            Formik - 
            <div
              className="Component-root-64"
            >
              Form - 
              <div
                component={[Function]}
                label="Full Name"
                name="fullname"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="Address"
                name="address"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="City"
                name="city"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="State / Province / Region"
                name="stateProvinceRegion"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="Postal Code / ZIP"
                name="zipcode"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="Clothing Size"
                name="clothingSize"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                component={[Function]}
                label="Shoe Size"
                name="shoeSize"
                type="text"
                validate={[Function]}
              >
                Field
              </div>
              <div
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              />
              <button
                className="MuiButtonBase-root-16 MuiButton-root-65 MuiButton-contained-76 MuiButton-containedPrimary-77 MuiButton-raised-79 MuiButton-raisedPrimary-80 MuiButton-fullWidth-90"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="submit"
              >
                <span
                  className="MuiButton-label-66"
                >
                  SHIP IT!
                </span>
                <span
                  className="MuiTouchRipple-root-91"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          className="Component-footer-4"
        >
          <a
            className="Component-footerLink-5"
            href="/terms_of_service"
            onClick={[Function]}
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-28 MuiTypography-caption-38 MuiTypography-colorTextSecondary-61 MuiTypography-alignCenter-51"
            >
              TERMS
            </span>
          </a>
          <a
            className="Component-footerLink-5"
            href="https://app.termly.io/document/privacy-policy/8192f107-a2dc-4548-bf83-97907c6d83c3"
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-28 MuiTypography-caption-38 MuiTypography-colorTextSecondary-61 MuiTypography-alignCenter-51"
            >
              PRIVACY
            </span>
          </a>
          <a
            className="Component-footerLink-5"
            href="/faq"
            onClick={[Function]}
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-28 MuiTypography-caption-38 MuiTypography-colorTextSecondary-61 MuiTypography-alignCenter-51"
            >
              FAIRNESS / FAQ
            </span>
          </a>
          <a
            className="Component-footerLink-5"
            href="/all-prizes"
            onClick={[Function]}
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-28 MuiTypography-caption-38 MuiTypography-colorTextSecondary-61 MuiTypography-alignCenter-51"
            >
              ALL PRIZES
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div>
    renderCreated - 
  </div>
</div>
`;
